etl/steps/data/garden/state_capacity/2023-11-10/information_capacity_dataset.py

```diff
@@ -0,0 +1,126 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import owid.catalog.processing as pr
+from owid.catalog import Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("information_capacity_dataset")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["information_capacity_dataset"].reset_index()
+
+    #
+    # Process data.
+    #
+
+    # Drop country id columns
+    tb = tb.drop(columns=["ccodecow", "vdemcode"])
+
+    # Harmonize country names.
+    tb = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+
+    # Reset index after dropping excluded countries
+    tb = tb.reset_index(drop=True)
+
+    # Add regional aggregations
+    tb = regional_aggregations(tb)
+
+    tb = tb.set_index(["country", "year"], verify_integrity=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir, tables=[tb], check_variables_metadata=True, default_metadata=ds_meadow.metadata
+    )
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def regional_aggregations(tb: Table) -> Table:
+    """
+    Add regional aggregations for some of the indicators
+    """
+
+    tb_regions = tb.copy()
+
+    # Load population data.
+    tb_pop = paths.load_dataset("population")
+    tb_pop = tb_pop["population"].reset_index()
+
+    # Merge population data.
+    tb_regions = tb_regions.merge(tb_pop[["country", "year", "population"]], how="left", on=["country", "year"])
+
```

Comment:
Instead of the previous, I'd suggest:
1. Loading all necessary datasets at the beginning of `run` (it's useful to clearly see what are the inputs of the step) instead of hidden inside the function. The function would take `tb_population` as an argument.
2. Use `data_helpers.geo.add_population_to_table` instead of manually merging your table with the population table.